airflow/utils/cli.py

```diff
@@ -140,10 +141,17 @@ def _build_metrics(func_name, namespace):
     :return: dict with metrics
     """
     sub_commands_to_check = {"users", "connections"}
+    sub_commands_to_check_based_upon_key = {"variables"}
     sensitive_fields = {"-p", "--password", "--conn-password"}
```

Comment:
Add a new structure called sensitive indexes and define that instead of hardcoding here

Reply:
Since this is very subjective to cli commands, Do we really need to take it out? I see the sub-commands mentioned here only for checking based upon flags (just the line above), to keep everything on same place, I wrote the variables subcommand here only and since this is specific to keys rather than flag, i used new variable.